File: Pattern/pattern.java
import java.io.*;
import java.util.*;

class Pattern {
     
static void printPattern(int r) {
 
    double dist;
 
    for (int i = 0; i <= 2 * r; i++) {
        // changed 'r' to '2 * r'
        for (int j = 0; j <= 2 * r; j++) {
            // changed '(j + r) * (j + r)' to '(j - r) * (j - r)'
            dist = Math.sqrt((i - r) * (i - r) + (j - r) * (j - r));
            if (dist > r - 0.5 && dist < r + 0.5)
            System.out.print("*");
            else
            System.out.print(" ");
        }
    
        System.out.print("\n");
    }
}
 
public static void main(String[] args)
    {   
        Scanner sc = new Scanner(System.in);
        int r;
        r = sc.nextInt();
        // changed 'printPattern()' to 'printPattern(r)'
        printPattern(r);
    }
}

File: Pattern/pattern.py
import math
 
def printPattern(r):
     
    for i in range((2 * r)+1):
        # changed 'r + 1' to '(2 * r)+1'
        for j in range((2 * r)+1):  
            # changed '(j + r) * (j + r)' to '(j - r) * (j - r)'
            dist = math.sqrt((i - r) * (i - r) + (j - r) * (j - r))
            if (dist > r - 0.5 and dist < r + 0.5):
                print("*",end="")
            else:
                print(" ",end="")     
        print()

r = int(input())
# changed 'printPattern()' to 'printPattern(r)'
printPattern(r)